brave/brave-core
# (Wallet) Add Filecoin support

## Resolves https://github.com/brave/brave-browser/issues/29258

## Details

* Adds support for Filecoin when creating a new account.
* Includes Filecoin flag to Android feature list retrieved from Core. (Disabled by default)
* Handles creation, renaming, and import of Filecoin accounts.
* **Update:** Shows a Filecoin network spinner to distinguish between mainnet and testnet. The spinner is shown _only_ for Filecoin type.

![Screenshot from 2023-04-14 14-02-20](https://user-images.githubusercontent.com/3308503/233018572-9b3232d9-19d8-4f1f-95de-3390949c04dc.png)

https://user-images.githubusercontent.com/3308503/233018809-8c6aaa88-82bc-4080-ae7f-760c69d26afb.mov

### [NEW] Filecoin Network spinner

![Screenshot_1682083777](https://user-images.githubusercontent.com/3308503/233648480-1c7746c9-7b35-46c2-ba67-9a19986295d6.png)

![Screenshot_1682083782](https://user-images.githubusercontent.com/3308503/233648519-3129d056-b3e2-4eae-ba2e-d4e383084f39.png)

## Security Review Requested

https://github.com/brave/security/issues/1256




## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan (**Disabled**):

_Creation_
* Navigate to Accounts section
* Tap on Add/Import account
* Select Filecoin
* Select Add
* Observe the new account has been added correctly to the primary list

_Renaming_
* Tap on the newly created Filecoin account
* Tap on Rename
* Type a new name for the account
* Tap Update
* Observe the new name is reflected correctly
* Tap back (top left corner)
* Observe the new name is reflected correctly in the primary list

_Import_
* Tap on the newly created Filecoin account
* Tap on Details
* Tap on Private Key
* Type the Wallet password and press continue
* Tap on the text "Copy to Clipboard" to save the private key (it will be used to restore the account)
* Tap back (top left corner) four times to navigate to the main Wallet screen
* Tap on the overrflow icon and select Settings
* Tap on "Reset Brave Wallet", then type `Yes`
* Follow the entire onboarding process to create a new Wallet
* Navigate to Accounts section
* Tap on Add/Import account
* Select Filecoin
* Paste the previously copied private key into the edit text
* Tap Add
* Observe the Filecoin account has been restored in the secondary list
